packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1780,6 +1782,7 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             this.emit("attached");
         }
         this.dataStores.setAttachState(attachState);
+        this.resetDirtyState(attachState);
```

Comment:
I do not think this call here is correct when we transition to "attached" state. Container maybe dirty, maybe not - it depends on whether it has pending ops or not.
Prior to that (detached and attaching state) it's dirty - that's fine.

Reply:
I just re-run couple of test cases again, and you are right about not needing to make this call on attach. Interestingly, there are no ops when container is attached. Meaning, the container summary is sent out without any `attach` ops. What triggers the reset of this flag is connection to delta manager (that coincidentally happens on  `attach`). Runtime tries to replay states (again, there are no pending ops), and subsequently clears the dirty flag. @vladsud Is this what you expected? In one way, it does make sense to flush on "connect", but I do find attach/connect/bind flow somewhat complex to reason about, and less cohesive. I'm still learning about it though :)

Reply:
Transitioning to attached state means file was created. At that moment we have file identity and can connect to relay service to send any ops. We might have some ops to send (that were generated while container was in attaching state) or not - that's not give. Just want to make sure it's clear that switch to attached state does not mean container is not dirty - it depends ("there are no pending ops" is incorrect in this context).
But I have not followed all the logic - it's possible it's fine to set it to be not dirty with assumption that logic that fires right after that due to "connected" event/transition will properly transition state to dirty if needed. That said, even if that's the case, it seems fishy that we would transition to "saved" state even for a blink of an eye, with pending changes. If possible, I'd prefer for "saved" state to be exposed to consumers only when there are truly no pending changes.

Reply:
Thanks for this clarification. I'll resolve this convo (as this call was removed) and look at the below comment around the flag init in constructor, keeping this use case in mind. Agree, we should be consistent: attached + pending ops = dirty (with no exceptions).